vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,352 @@
+	if adminOrg.AdminOrg == nil || adminOrg.AdminOrg.HREF == "" {
+		return nil, nil, fmt.Errorf("error retrieving org %s", d.Get("org").(string))
+	}
+	userName := d.Get("name").(string)
+	orgUser, err := adminOrg.FetchUserByName(userName, false)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return orgUser, &adminOrg, nil
+}
+
+func setOrgUserData(d *schema.ResourceData, orgUser *govcd.OrgUser, adminOrg *govcd.AdminOrg) error {
+
+	d.SetId(orgUser.User.ID)
+	err := d.Set("org", adminOrg.AdminOrg.Name)
+	if err != nil {
+		return err
+	}
+	err = d.Set("name", orgUser.User.Name)
+	if err != nil {
+		return err
+	}
+	err = d.Set("provider_type", orgUser.User.ProviderType)
+	if err != nil {
+		return err
+	}
+	err = d.Set("is_group_role", orgUser.User.IsGroupRole)
+	if err != nil {
+		return err
+	}
+	err = d.Set("description", orgUser.User.Description)
+	if err != nil {
+		return err
+	}
+	err = d.Set("full_name", orgUser.User.FullName)
+	if err != nil {
+		return err
+	}
+	err = d.Set("email_address", orgUser.User.EmailAddress)
+	if err != nil {
+		return err
+	}
+	err = d.Set("telephone", orgUser.User.Telephone)
+	if err != nil {
+		return err
+	}
+	err = d.Set("instant_messaging", orgUser.User.IM)
+	if err != nil {
+		return err
+	}
+	err = d.Set("is_enabled", orgUser.User.IsEnabled)
+	if err != nil {
+		return err
+	}
+	err = d.Set("is_locked", orgUser.User.IsLocked)
+	if err != nil {
+		return err
+	}
+	err = d.Set("deployed_vm_quota", orgUser.User.DeployedVmQuota)
+	if err != nil {
+		return err
+	}
+	err = d.Set("stored_vm_quota", orgUser.User.StoredVmQuota)
+	if err != nil {
+		return err
+	}
+	err = d.Set("role", orgUser.User.Role.Name)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceVcdOrgUserCreate(d *schema.ResourceData, meta interface{}) error {
+
+	userData, adminOrg, err := resourceToUserData(d, meta)
+	if err != nil {
+		return err
+	}
+	_, err = adminOrg.CreateUserSimple(*userData)
+	if err != nil {
+		return err
+	}
+	return resourceVcdOrgUserRead(d, meta)
+}
+
+func resourceVcdOrgUserDelete(d *schema.ResourceData, meta interface{}) error {
+
+	takeOwnership := d.Get("take_ownership").(bool)
+	orgUser, _, err := resourceToOrgUser(d, meta)
+	if err != nil {
+		return err
+	}
+	return orgUser.Delete(takeOwnership)
+}
+
+func resourceVcdOrgUserRead(d *schema.ResourceData, meta interface{}) error {
+
+	orgUser, adminOrg, err := resourceToOrgUser(d, meta)
+	if err != nil {
+		return err
+	}
+	return setOrgUserData(d, orgUser, adminOrg)
+}
+
+func resourceVcdOrgUserUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	orgUser, _, err := resourceToOrgUser(d, meta)
+	if err != nil {
+		return err
+	}
+	userData, _, err := resourceToUserData(d, meta)
+	if err != nil {
+		return err
+	}
+	err = orgUser.UpdateSimple(*userData)
+	if err != nil {
+		return err
+	}
+	return resourceVcdOrgUserRead(d, meta)
+}
+
+func resourceVcdOrgUserImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
Please add a beefy function comment - similar to the ones we have for LB imports.

Reply:
done

Reply:
done